### ISSUE TITLE
AWS Transfer Family support for Amazon Elastic File System

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#17022]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> AWS Transfer Family now supports file transfers for Amazon Elastic File System (Amazon EFS) file systems.

### New or Affected Resource(s)

* [`aws_transfer_server`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_server)
* [`aws_transfer_user`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_user)

### Potential Terraform Configuration

```hcl
resource "aws_transfer_server" "example" {
  domain = "EFS"
}

resource "aws_transfer_user" "example" {
  posix_profile {
    uid = 1000
    gid = 1000
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2021/01/introducing-aws-transfer-family-support-for-amazon-efs/).
[Blog post](https://aws.amazon.com/blogs/aws/new-aws-transfer-family-support-for-amazon-elastic-file-system/).

Requires [AWS SDK v1.36.22](https://github.com/aws/aws-sdk-go/releases/tag/v1.36.22):

* https://github.com/hashicorp/terraform-provider-aws/pull/17013

**Comment by larstobi** [comment]
AWS SDK v1.36.28 has been merged into terraform-provider-aws (#17155).

**Comment by spaceygithub** [comment]
Hello, can you please confirm if this made it in to aws provider 3.25.0 release?
I'm trying v3.37.0 and currently getting 
```
Error: Unsupported argument

  on transfer_server.tf line 42, in resource "aws_transfer_server" "this":
  42:   domain       = "EFS"
```
I also see no mention of EFS on https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_server
